secret/secrets_manager_client_test.go

```diff
@@ -61,11 +72,77 @@ func TestSecretsManagerCreateAndDeleteSecret(t *testing.T) {
 		require.NoError(t, err)
 		require.NotZero(t, out)
 
+		defer Cleanup(out)
+
+		require.NotZero(t, out.Name)
+		require.NotZero(t, out.ARN)
+
+	})
+
+	t.Run("GetFailsWithInvalidInput", func(t *testing.T) {
+		out, err := c.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{})
+		assert.Error(t, err)
+		assert.Zero(t, out)
+	})
+
+	t.Run("UpdateFailsWithInvalidInput", func(t *testing.T) {
+		out, err := c.UpdateSecret(ctx, &secretsmanager.UpdateSecretInput{})
+		assert.Error(t, err)
+		assert.Zero(t, out)
+	})
+
+	t.Run("CreateAndGetSucceed", func(t *testing.T) {
+		out, err := c.CreateSecret(ctx, &secretsmanager.CreateSecretInput{
+			Name:         aws.String(os.Getenv("AWS_SECRET_PREFIX") + "foobar"),
+			SecretString: aws.String("barfoo"),
+		})
+		require.NoError(t, err)
+		require.NotZero(t, out)
+
+		defer Cleanup(out)
+
+		defer func() {
+			if out != nil && out.Name != nil && out.ARN != nil {
+				out, err := c.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{
+					SecretId: out.ARN,
+				})
+				require.NoError(t, err)
+				require.NotZero(t, out)
+				assert.Equal(t, "barfoo", *out.SecretString)
+				assert.Equal(t, os.Getenv("AWS_SECRET_PREFIX")+"foobar", *out.Name)
+			}
+		}()
+
+		require.NotZero(t, out.Name)
+		require.NotZero(t, out.ARN)
+	})
+
+	t.Run("CreateAndUpdateAndGetSucceed", func(t *testing.T) {
```

Comment:
I would rename this to "UpdateSecretModifiesValue" - tests are easier to understand when they can be summed up nicely as what you're trying to verify rather than the code that you're calling.